README.md

```diff
@@ -2,43 +2,145 @@
+
+wolfSentry is designed from the ground up to function well in
+resource-constrained, bare-metal, and realtime environments, with algorithms to
+stay within designated maximum memory footprints and maintain deterministic
+throughput.  This allows full firewall and IDPS functionality on embedded
+targets such as FreeRTOS, Nucleus, NUTTX, Zephyr, VxWorks, and Green Hills
+Integrity, and on ARM and other common embedded CPUs and MCUs.  wolfSentry with
+dynamic firewalling can add as little as 64k to the code footprint, and 32k to
+the volatile state footprint, and can fully leverage the existing logic and
+state of applications and sibling libraries.
 
 
 ## Documentation
 
-Basic application integration on FreeRTOS-lwIP is documented, with usable code fragments, by [doc/freertos-lwip-app.md](doc/freertos-lwip-app.md).
+Basic application integration on FreeRTOS-lwIP is documented, with usable code fragments, in [doc/freertos-lwip-app.md](doc/freertos-lwip-app.md).
 
-The JSON configuration blob is documented in detail by [doc/json_configuration.md](doc/json_configuration.md).
+JSON configuration is documented in detail by [doc/json_configuration.md](doc/json_configuration.md).
 
+The full programmer's manual in PDF form is available at [doc/wolfSentry_refman.pdf](doc/wolfSentry_refman.pdf).
+
+With `doxygen` installed, the HTML version of the reference manual can be
+generated with `make doc-html`.
 
 ## Dependencies
 
 In its default build, wolfSentry depends on a POSIX runtime, specifically the
 heap allocator, clock_gettime, stdio, semaphore, pthreads, and string APIs.
-However, these dependencies can be avoided with various build-time options.  In
-particular, the recipe
-
+However, these dependencies can be avoided with various build-time options.  The recipe
+<br>
 ```
 make STATIC=1 SINGLETHREADED=1 NO_STDIO=1 EXTRA_CFLAGS='-DWOLFSENTRY_NO_CLOCK_BUILTIN -DWOLFSENTRY_NO_MALLOC_BUILTIN'
 ```
 
-generates a libwolfsentry.a that depends on only a handful of basic string
-functions.  Allocator and time callbacks must then be set in a `struct
-wolfsentry_host_platform_interface` supplied to `wolfsentry_init()`.
-
-
-## Building and testing
-
-Build and test libwolfsentry.a on Linux:
+builds a libwolfsentry.a that depends on only a handful of basic string
+functions and the `inet_ntop()` library function (from POSIX.1-2001, and also
+implemented by lwIP).  Allocator and time callbacks must then be set in a
+`struct wolfsentry_host_platform_interface` supplied to `wolfsentry_init()`.
+
+The wolfSentry `Makefile` depends on a modern (v4.0+) Gnu `make`.  The library
+itself can be built outside `make`, within another project/framework, by
+creating a user settings macro file and passing its path to the compiler with
+the `WOLFSENTRY_USER_SETTINGS_FILE` macro.
+
+
+## Building
+
+wolfSentry was written with portability in mind, with provisions for non-POSIX
+and C89 targets.  For example, all its dependencies can be met with the
```

Comment:
double space between '.' and 'For'

Reply:
Is it intentional, looks like between '.' and 'If' in the next sentence and several other locations?

Reply:
Yeah there's gonna be double spaces after periods all through that codebase.  Long ago this was the norm btw!